### DISCUSSION TITLE
Generation of physical groups for meshes

### DISCUSSION THREAD

**Discussion by JanSuntajs** [neperfepx/neper#802] [Q&A] [answered]
Hello, 

I have just started including Neper into my workflow for simulations of PDEs on multimaterial domains. Up untill now, 
I've been testing my code on homogenous (single-material or single-cell) domains such as spheres, cubes, cylinders etc. I've generated meshes for these structures using the [OCC kernel of Gmsh](https://gmsh.info/doc/texinfo/gmsh.html#t16). For later use and definition of the problem's PDEs, I like to define physical regions with `PhysicalNames` in `Gmsh`; for a sphere, those would typically be named `Casing`  (the sphere's surface, typically used for the specification of the boundary conditions), `Center` (a point at the center, since sometimes a special boundary condition needs to be imposed, such as the zero-displacement BC) and `Volume`:
```msh
$MeshFormat
4.1 0 8
$EndMeshFormat
$PhysicalNames
3
0 1 "Center"
2 1 "Casing"
3 1 "Volume"
$EndPhysicalNames
$Entities
3 3 1 1
1 6.735557395310444e-17 -1.349783804395672e-32 0.8 0 
2 6.735557395310444e-17 -1.349783804395672e-32 -0.8 0 
3 0 0 0 1 1 
1 -1.100000005052893e-07 -1.10000000588556e-07 0.7999998899999995 1.100000004605604e-07 1.100000005775257e-07 0.8000001100000005 0 2 1 -1 
2 -1.100000002196566e-07 -1.100000002204364e-07 -0.8000001100000003 1.10000011 1.1e-07 0.80000011 0 2 2 -1 
3 -1.100000004810976e-07 -1.100000003330679e-07 -0.8000001100000004 1.100000004440902e-07 1.100000004070828e-07 -0.7999998899999993 0 2 2 -2 
1 -2.20000011 -1.55884583681199 -0.80000011 1.10000011 1.558845836811989 0.80000011 1 -1 4 -1 -2 3 2 
1 -2.20000011 -1.55884583681199 -0.80000011 1.10000011 1.558845836811989 0.80000011 1 1 1 1 
$EndEntities

``` 

I am wondering, if something similar is also possible inside Neper and with a polycrystalline sample in mind. Additionally, I would also like to know whether it's possible to group entities belonging to different grains (cells) in a similar manner. What I am getting right now does not produce the desired results entirely. Namely, each entity, be it a point, edge or a face, belongs to its own physical group, as shown in the second appended `.msh` file:
```msh

$PhysicalNames
1645
0 1 "ver1"
0 2 "ver2"
0 3 "ver3"
0 4 "ver4"
0 5 "ver5"
0 6 "ver6"
0 7 "ver7"
0 8 "ver8"
0 9 "ver9"
0 10 "ver10"
0 11 "ver11"
0 12 "ver12"
0 13 "ver13"
0 14 "ver14"
0 15 "ver15"
0 16 "ver16"
0 17 "ver17"
0 18 "ver18"
0 19 "ver19"
0 20 "ver20"
0 21 "ver21"
0 22 "ver22"
0 23 "ver23"
0 24 "ver24"
0 25 "ver25"
0 26 "ver26"
0 27 "ver27"
# and so forth for edges and faces
 ```

The commands I am currently using are the following:
```bash
neper -T -n 100 -domain "sphere(1, 100):translate(-0.5,-0.5,-0.5)" -morpho "gg" -dim "3" -group "id" -o "sphere"
neper -M sphere.tess
``` 

I apologise if this duplicates any of the similar problems, I was unsure if those ([504](https://github.com/neperfepx/neper/discussions/504?fbclid=IwAR0Tjp9iydXmSDfkDRwsyss4-bysS-r-G6Z1HpSTOx8YEMKqDzL00xquMP4), [754](https://github.com/neperfepx/neper/discussions/754?fbclid=IwAR1nosHUftT1VzzjCiYJokgsF2ASWTqghM2sGQdxq09yNOjtm5hXaYU-K2M)) apply to the same use case
Regards,
Jan

**Comment by rquey** [comment, marked as answer]
This discussion has been closed as obsolete. Please repost if necessary.
